src/ServicePulse.Host/app/js/views/archive/controller.js

```diff
@@ -72,15 +73,29 @@
             vm.loadingData = false;
         };
 
+        var storeGroupOption = function (amount, unit) {
+            $cookies.put('archive_amount', amount);
+            $cookies.put('archive_unit', unit);
+        }
+
+        var loadDefaultGroup = function() {
+            var amount = $cookies.get("archive_amount");
+            var unit = $cookies.get("archive_unit");
+
+            vm.timeGroup.amount = amount;
+            vm.timeGroup.unit = unit;
+        }
+
         var init = function () {
 
             vm.configuration = sharedDataService.getConfiguration();
             vm.error_retention_period = moment.duration(vm.configuration.data_retention.error_retention_period).asHours();
             vm.total = 1;
             vm.archives = [];
             vm.sort.page = 1;
-            vm.sort.start = moment.utc().subtract(vm.timeGroup.amount, vm.timeGroup.unit).format('YYYY-MM-DDTHH:mm:ss');
-            vm.sort.end = moment.utc().format('YYYY-MM-DDTHH:mm:ss');
+
+            loadDefaultGroup();
```

Comment:
See above, it feels hidden as to what the `loadDefaultGroup` method does. Once it's loaded the default group, there's no indication in code that it sets the values to the `timeGroup` variable making it seem like there's no link between the load and the use of the timeGroup variable, someone might think it's ok to swap the lines around.